cmd/kubectl-gadget/bcck8s.go

```diff
@@ -226,6 +227,14 @@ func bccCmd(subCommand, bccScript string) func(*cobra.Command, []string) {
 			podnameFilter = fmt.Sprintf("--podname %q", podnameParam)
 		}
 
+		containernameFilter := ""
+		if containernameParam != "" {
+			if podnameParam == "" {
+				contextLogger.Fatalf("--containername must be used with --podname")
```

Comment:
Is there a reason for this restriction? I imagine that for deployments, the pod name is not known in advance, so a common use case could be to use `--label` with `--containername`.

Reply:
I wanted to keep it similar to `kubectl exec $POD -c $CONTAINER` but I just realized it doesn't make any sense and the use case you mention about deployments makes a lot of sense. I removed it.